src/metax_api/permissions/permissions.py

```diff
@@ -146,10 +143,13 @@ def _check_rpc_perms(self, request, api_name):
         return has_perm
 
     def has_object_permission(self, request, view, obj):
-        has_perm = obj.user_has_access(request)
-
-        if not has_perm:
-            self.message = 'You are not permitted to access this resource.'
+        has_perm = False
+        if 'all' in self.perms['rest'][view.get_api_name()].get(METHOD_MAP[request.method], []):
+            has_perm = True
+        else:
+            has_perm = obj.user_has_access(request)
+            if not has_perm:
+                self.message = 'You are not permitted to access this resource.'
```

Comment:
I think this will not work with dataset drafts? Since read operations to /rest/datasets is defined as "all" in API_ACCESS in app_config, this would bypass checking if object is a draft and whether current user has permission to view it or not.

What about keeping permissions.py as it was, and instead defining ``user_has_access()`` method in DataCatalog model, and always return true for read operations? Unless there is yet another related issue that I'm missing.

Reply:
I was thinking to make permission checking to be more general and didn't consider drafts. Your proposition does indeed work for data catalogs so will do new pr.